internal/server/data/destination.go

```diff
@@ -4,30 +4,67 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/infrahq/infra/internal"
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/uid"
 )
 
+type destinationsTable models.Destination
+
+func (d destinationsTable) Table() string {
+	return "destinations"
+}
+
+func (d destinationsTable) Columns() []string {
+	return []string{"connection_ca", "connection_url", "created_at", "deleted_at", "id", "last_seen_at", "name", "organization_id", "resources", "roles", "unique_id", "updated_at", "version"}
+}
+
+func (d destinationsTable) Values() []any {
+	return []any{d.ConnectionCA, d.ConnectionURL, d.CreatedAt, d.DeletedAt, d.ID, d.LastSeenAt, d.Name, d.OrganizationID, d.Resources, d.Roles, d.UniqueID, d.UpdatedAt, d.Version}
+}
+
+func (d *destinationsTable) ScanFields() []any {
+	return []any{&d.ConnectionCA, &d.ConnectionURL, &d.CreatedAt, &d.DeletedAt, &d.ID, &d.LastSeenAt, &d.Name, &d.OrganizationID, &d.Resources, &d.Roles, &d.UniqueID, &d.UpdatedAt, &d.Version}
+}
+
+// destinationsUpdateTable is used to update the destination. It excludes
+// the CreatedAt field, because that field is not part of the input to
+// UpdateDestination.
+type destinationsUpdateTable models.Destination
```

Comment:
This feels a bit off, but I can't come up with a better way to do updates either in our current model

Reply:
It is unfortunate, there are some option options:

1. We could `select created_at` and set the value on the struct, before the `UPDATE`
2. We could not use `insert()` and copy out the implementation into `CreateDestination`. 

Maybe option 2 is better, I'll see what it looks like

Reply:
I guess the problem with 2 is that we have to add `CreatedAt` back in for both `Get` and `List`, which means we end up having to deal with the special case in 3 places, instead of only `Update` being a special case.